cloudpickle/cloudpickle.py

```diff
@@ -187,32 +195,6 @@ def _builtin_type(name):
     return getattr(types, name)
 
 
-def _make__new__factory(type_):
-    def _factory():
-        return type_.__new__
-    return _factory
-
-
-# NOTE: These need to be module globals so that they're pickleable as globals.
-_get_dict_new = _make__new__factory(dict)
-_get_frozenset_new = _make__new__factory(frozenset)
-_get_list_new = _make__new__factory(list)
-_get_set_new = _make__new__factory(set)
-_get_tuple_new = _make__new__factory(tuple)
-_get_object_new = _make__new__factory(object)
```

Comment:
I would have thought that keeping those symbols in the cloudpickle module would have been necessary to be able to unpickle pickle strings generated by versions of cloudpickle older than <= 1.1.0 but it does not seem to be the case.

Reply:
I had not thought about this, and now that you said it, your conclusion surprises me. On what object did you try it?

Reply:
Now that you say it, I wonder if those functions ever worked as their were intended to work.
`cloudpickle._get_dict_new` is actually not a "global" but a nested function. Thus, such constructors are actually pickled as dynamic. As those constructors are recreated at depickling time, this code was forward compatible (hence no failures) :smile: